scipy/linalg/flapack.pyf.src

```diff
@@ -735,11 +763,296 @@ interface
           :: lwork=2*minmn+MAX(maxmn,nrhs)
           ! check(lwork>=2*minmn+MAX(maxmn,nrhs))
```

Comment:
Allowing -1 in these routines is not good, since you then copy big arrays to fortran-order buffers just for the lwork computation if the input is in c-order. There's a separate `gelss_lwork` for that calculation above.

The best option really would be if the default was to obtain lwork from lapack, but it's not done in these f2py wrappers. It seems that this option is made impossible to do by the fact that some of these routines return also the work arrays, apparently in a misguided attempt to make the lwork computations work.

The better way to go, rather than attempting to fix this f2py stuff which is constrained by backward compatibility considerations, is to concentrate on the new Cython interfaces and try to do them better.